File: SIC_Backend/Data/Models/RegisterModel.cs

using System.ComponentModel.DataAnnotations;


namespace SIC_Backend.Data.Models;

public class RegisterModel
{
    [Required]
    public required string FirstName { get; set; }

    [Required]
    public required string LastName { get; set; }

    [Required]
    [EmailAddress]
    public required string Email { get; set; }

    [Required]
    public required string Username { get; set; }

    [Required]
    [DataType(DataType.Password)]
    public required string Password { get; set; }

}

File: sic_home/lib/models/authentication/register_model.dart
class RegisterModel {
  String firstName;
  String lastName;
  String email;
  String username;
  bool gender;
  String password;

  RegisterModel({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.username,
    required this.gender,
    required this.password,
  });

  factory RegisterModel.fromJson(Map<String, dynamic> json) {
    return RegisterModel(
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      username: json['username'],
      gender: json['gender'],
      password: json['password'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'username': username,
      'gender': gender,
      'password': password,
    };
  }
}
